e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js

```diff
@@ -2610,3 +2614,55 @@ describe("issue 32499", () => {
     }
   });
 });
+
+describe("issue 23449", () => {
+  beforeEach(() => {
+    H.restore();
+    cy.signInAsAdmin();
+  });
+
+  it("Remapped fields should work in a model", () => {
+    // set the metadata for review
+    cy.visit(
+      `/admin/datamodel/database/${SAMPLE_DB_ID}/schema/${SAMPLE_DB_SCHEMA_ID}/table/${SAMPLE_DATABASE.REVIEWS_ID}`,
+    );
+
+    cy.findByTestId("column-RATING").findByLabelText("Field settings").click();
+    cy.findAllByTestId("select-button").contains("Use original value").click();
+    cy.findByLabelText("Custom mapping").click();
+
+    const values = { 1: "A", 2: "B", 3: "C", 4: "D", 5: "E" };
```

Comment:
I know this works but having numbers as object literal keys is a bit weird.

You could write `{ "A": 1, ... }`, but maybe even more straightforward would be to just unfurl the loop altogether:

```suggestion
    cy.findByDisplayValue("1").clear().type("A");
    cy.findByDisplayValue("2").clear().type("B");
```